app/Models/Map.php

```diff
@@ -728,60 +728,28 @@ public function hasDistanceUnit(): bool
      */
     public function datagridActions(Campaign $campaign): array
     {
-        $actions = [];
+        $newActions = [];
+        $actions = parent::datagridActions($campaign);
 
-        // Relations & Inventory
-        if (!isset($this->hasRelations)) {
-            $actions[] = '<li>
-                <a href="' . route('entities.relations.index', $this->entity) . '" class="dropdown-item datagrid-dropdown-item" data-name="relations">
-                    <i class="fa-solid fa-users" aria-hidden="true"></i> ' . __('crud.tabs.connections') . '
-                </a>
-            </li>';
-
-            if ($campaign->enabled('inventories')) {
-                $actions[] = '<li>
-                <a href="' . route('entities.inventory', $this->entity) . '" class="dropdown-item datagrid-dropdown-item" data-name="inventory">
-                    <i class="ra ra-round-bottom-flask" aria-hidden="true"></i> ' . __('crud.tabs.inventory') . '
-                </a>
-            </li>';
-            }
-
-            if ($campaign->enabled('abilities') && $this->entityTypeId() != config('entities.ids.ability')) {
-                $actions[] = '<li>
-                <a href="' . route('entities.entity_abilities.index', $this->entity) . '" class="dropdown-item datagrid-dropdown-item" data-name="abilities">
-                    <i class="ra ra-fire-symbol" aria-hidden="true"></i> ' . __('crud.tabs.abilities') . '
-                </a>
-            </li>';
-            }
-            if ($campaign->enabled('maps') && $this->entityTypeId() == config('entities.ids.map')) {
-                $actions[] = '<li>
+        if ($campaign->enabled('maps') && $this->entityTypeId() == config('entities.ids.map') && auth()->check() && auth()->user()->can('update', $this)) {
```

Comment:
We know we're on a map, so the map module is always enabled, and the entity type is already a map, so we only need to check for the permission to update the entity ;-)